fusillade/clouddirectory.py

```diff
@@ -1313,15 +1327,67 @@ def add_roles(self, roles: List[str]):
     def remove_roles(self, roles: List[str]):
         operations = []
         operations.extend(self._remove_links_batch(roles, Role.object_type))
-        operations.extend(self._remove_typed_links_batch(roles, Role.object_type))
+        operations.extend(self._remove_typed_links_batch(roles,
+                                                         Role.object_type,
+                                                         'membership_link',
+                                                         {'member_of': Role.object_type}))
         self.cd.batch_write(operations)
         self._roles = None  # update roles
         logger.info(dict(message="Roles removed",
                          object=dict(type=self.object_type, path_name=self._path_name),
                          roles=roles))
 
 
-class User(CloudNode, RolesMixin):
+class OwnershipMixin:
+    ownable = ['group', 'role']
+
+    def add_ownership(self, node: Type['CloudNode']):
+        self.cd.attach_typed_link(
+            self.object_ref,
+            node.object_ref,
+            'ownership_link',
+            {'owner_of': node.object_type})
+
+    def remove_ownership(self, node: Type['CloudNode']):
+        typed_link_specifier = self.cd.make_typed_link_specifier(
+            self.object_ref,
+            node.object_ref,
+            'ownership_link',
+            {'owner_of': node.object_type}
+        )
+        self.cd.detach_typed_link(typed_link_specifier)
+
+    def is_owner(self, node: Type['CloudNode']):
+        tls = self.cd.make_typed_link_specifier(
+            self.object_ref,
+            node.object_ref,
+            'ownership_link',
+            {'owner_of': node.object_type})
+        try:
+            self.cd.get_link_attributes(tls, [])
```

Comment:
If this succeeds, it looks like it returns `None`, is that intended?

Reply:
yes, otherwise it will return resource not found. We just need to check that it exists. This the easiest way the cloud-directory API lets me without list all incoming or outgoing typed links.

Reply:
Cool, this PR looks good then.  The rest are nits.